### DISCUSSION TITLE
Any example to pass a directory for -f options?

### DISCUSSION THREAD

**Discussion by hdw868** [locustio/locust#2991] [Q&A]
As described in https://docs.locust.io/en/stable/configuration.html

Command line | Environment | Config file | Description
-- | -- | -- | --
-f, --locustfile | LOCUST_LOCUSTFILE | locustfile | The Python file or module that contains your test, e.g. ‘my_test.py’. Accepts multiple comma-separated .py files, a package name/directory or a url to a remote locustfile. Defaults to ‘locustfile’.

I tried to pass a directory as locustfile option in master node:
```bash
locust -f test_cases/ --master --class-picker
```
on the worker node, when I try to launch worker with  `-f -`option, it reports:
```bash
python3 -m locust -f - --worker --master-host my_host --processes -1
Got error from master: locustfile parameter on master must be a plain filename (not a module name) (was 'test_cases/')
```
Any idea on what's happening?

**Comment by cyberw** [comment]
Only plain filenames are supported for locustfile distribution to workers (if someone wants to build it then that would be great, but I think it would be complicated). 

I'll have a quick look at making the error message more clear.

**Reply by hdw868** [reply]
Thanks for the quick response, it does not support the directory in fact?
The problem is if your locustfile import something from other modules, then it will raise import exception because the other module file is not transported

**Comment by cyberw** [comment]
What version of locust are you on btw, maybe it has already been fixed? At least I cannot find that specific error message anywhere in the code.

**Reply by hdw868** [reply]
Locust 2.25.0

**Reply by cyberw** [reply]
that's really old...
